### DISCUSSION TITLE
/app i18n sitemap

### DISCUSSION THREAD

**Discussion by Rover420** [vercel/next.js#53079] [App Router]
Hi. 

I would want to ask if this is a correct approach to generate sitemap for my app with /app dir and i18n or maybe is there so built-in sitemap generation in Next 13? Or should I just generate a sitemap without the locale, because if someone would go to my page without locale path, he would be redirected to specific locale page anyway with middleware...

// /app/sitemap.js

export default function sitemap() {
    const locale = ['pl', 'en']

    return locale.flatMap(item => [
            {
                url: `https://mywebsite.vercel.app/${item}`,
                lastModified: new Date(),
            },
            {
                url: `https://mywebsite.app/${item}/register`,
                lastModified: new Date(),
            },
            {
                url: `https://mywebsite.app/${item}/signin`,
                lastModified: new Date(),
            },
        ]
    )
}

output:

![image](https://github.com/vercel/next.js/assets/100524342/1d69cd3e-c906-4f24-8a31-ba886b02a615)

Thanks for help

**Comment by pschaub** [comment]
There is an official solution how to handle this but sadly Next.js currently doesn't support this out-of-the-box.
See https://developers.google.com/search/docs/specialty/international/localized-versions?hl=en#sitemap
